analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+  ) |>
+  # sum all values for each
+  dplyr::group_by(barcodes) |>
+  dplyr::summarize(total_cnv_per_cell = sum(cnv)) |>
+  dplyr::ungroup() |>
+  # bring back metadata
+  dplyr::inner_join(
+    infercnv_metadata_df,
+    by = "barcodes"
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df)
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+ggplot(has_cnv_df) +
+  aes(x = total_cnv_per_cell, fill = cell_group) +
+  geom_density(alpha = 0.5) +
+  theme(legend.position = "bottom")
+```
+
+### CNVs across cell types
+
+Next, we'll look at the per-cell distributions of CNVs across cell types, using labels derived from the `cell-type-ewings` module as follows:
+
+* `tumor-endo`: Cells labeled as tumor by cell-type-ewings, but endothelial by `cell-type-consensus`
+* `tumor-immune`: Cells labeled as tumor by cell-type-ewings, but immune by `cell-type-consensus`
+* `tumor`: Cells labeled as tumor by `cell-type-ewings`
+* `endo`: An endothelial cell type
+* `immune`: An immune cell type
+* If none of the above, their existing `cell-type-ewings` annotation
+
+Ideally, we'll see that tumor and Unknown cells from will have relatively higher CNV scores compared to other cell types, in particular those cell types in the reference.
+
+```{r fig.height=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
+query_has_cnv_df <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id)
+
+ggplot(query_has_cnv_df) +
+  aes(
+    y = celltype,
+    x = total_cnv_per_cell,
+    fill = cell_group
+  ) +
+  ggridges::geom_density_ridges2(alpha = 0.7, scale = 0.85)
+```
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+Below, the UMAP colored by cell type is shown on the left, and the UMAP colored by total CNV counts is shown on the right.
+Ideally, unknown and tumor cells will overlap with regions in the UMAP with higher total CNV.
+
+
+```{r, fig.height = 10}
+celltype_umap <- ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = celltype) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  labs(title = "Cell types") +
+  umap_theme
+
+cnv_umap <- ggplot(query_has_cnv_df) +
+  aes(x = UMAP1, y = UMAP2, color = total_cnv_per_cell) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  scale_color_viridis_c() +
+  labs(title = "Total CNV per cell") +
+  umap_theme
+
+celltype_umap / cnv_umap
+```
+
+
+## Clusters
+
+Part of `inferCNV`'s algorithm involves clustering the data, and poor clusters have the potential to lead to inaccurate inferences of CNV.
+To see whether we'd like to further tune the clustering parameters, we'll get a sense of their size and rough composition here.
+
+How many clusters are there, and how big are they?
+We'll show this separately for reference vs query cells.
+
+```{r}
+cluster_count_df <- has_cnv_df |>
+  dplyr::count(subcluster, cell_group)
+ggplot(cluster_count_df) +
+  aes(x = n, fill = cell_group) +
```

Comment:
I'm confused what n is here? Is it the number of cells in a cluster and the x axis is the number of clusters with that same count? Can you update the labels to be more specific?